admin/forms.py

```diff
@@ -1,17 +1,44 @@
 from flask_wtf import FlaskForm
 from wtforms import StringField, SelectField, PasswordField, SubmitField
-from wtforms.validators import DataRequired, Email, Length
+from wtforms.validators import DataRequired, Regexp, Length, ValidationError
 from wtforms.fields.html5 import EmailField
 
 from models import Role, Status, User
 
+EMAIL_PATTERN = r'^(\w|\.|\_|\-)+[@](\w|\_|\-|\.)+[.]\w{2,3}$'
+
+
+def unique_check(column):
+    message = f'{column.title()} is already taken'
+
+    def _unique_check(form, field):
+        if User.objects(__raw__={column: field.data}).first():
+            raise ValidationError(message)
+
+    return _unique_check
+
+
+def update_unique_check(column):
+    message = f'{column.title()} is already taken'
+
+    def _update_unique_check(form, field):
+        user = User.objects(__raw__={column: field.data}).first()
+        if user and user.id != form['user_id'].data:
+            raise ValidationError(message)
+
+    return _update_unique_check
+
```

Comment:
We may change this two functions into one, adhere to DRY principle:

`def unique_check(column, update=False):
    message = f'{column.title()} is already taken'

    def _unique_check(form, field):
        user = User.objects(__raw__={column: field.data}).first()

        if update and user and user.id != form['user_id'].data:
            raise ValidationError(message)

        if user and not update:
            raise ValidationError(message)

        return _unique_check`